### DISCUSSION TITLE
[NUXT 3] How to exclude files from being watched by the dev server?

### DISCUSSION THREAD

**Discussion by tintin10q** [nuxt/nuxt#17328] [Questions]
Hi thanks for taking the time to read this, 

I am using better-sqlite3 in my nuxt 3 project and every time I update something in the database (storage.db in the root of the project), the dev server counts this as a file change and rebuilds the server. This actually causes errors because the dev server is unavailable for a short time (200ms) and any requests made to it return a 404 during this time. This makes the dev server basically useless to me as many things fail which won't fail when using the production server (because it does not rebuild). I don't want to move my database file from the root of the project as this introduces more complexity as to how to locate it as everything executes from the root of the directory. 

Therefore I ask: **How can I exclude some files from being watched by the dev server?** I have already tried the `.nuxtignore` file but that did not work. Is it vite config? Something in the nuxt config? I hope someone has an answer for this.

**Comment by viii47** [comment]
Vite has a new rollup option to ignore specific files or folders (see https://github.com/nuxt/framework/pull/5632)

**Reply by tintin10q** [reply]
Thank you! I will try this. What you linked to indicates to me that the contents of `.nuxtignore` will be automatically added to the rollup option? Is that feature already released?

**Reply by viii47** [reply]
> Thank you! I will try this. What you linked to indicates to me that the contents of `.nuxtignore` will be automatically added to the rollup option? Is that feature already released?

The latest RC is from 24 days ago, that commit with the vite extention is 10 days old so it's not included, however if you want to have a look you can opt into the edge-channel (newest commit that passed all tests) - documentation for this is here: https://v3.nuxtjs.org/guide/going-further/edge-channel  😊

**Reply by tintin10q** [reply]
Thanks!

It does not seem to work for me however. 

This is my `rollup.config.js`
```
export default {
  watch: {
    exclude: ['storage.db','node_modules/**']
  }
};
```

It still rebuilds the dev server however on every insert. 

Here is my project: [https://github.com/tintin10q/nuxt3-sqlite-template](https://github.com/tintin10q/nuxt3-sqlite-template)

**Comment by JBustin** [comment]
Hello
I'm also trying to exclude files from watchers. Did you find any solution ?
I just try with nitro configuration in `nuxt.config.ts`, but unsuccessfully:
```
nitro: {
  watchOptions: {
    ignore: file => shouldNotBeWatched(file)
  }
}
```

**Reply by sewalsh** [reply]
`nuxt.config.ts`

```ts filename="nuxt.config.ts"
vite: {
    server: {
      watch: {
        // Exclude specific files or directories from being watched
        ignored: [
          '**/pages/**',
          '**/pages/index.vue',
        ],
      },
    },
  },
```
